### PR TITLE
🧙‍♂️ Handle service injection in async function

### PR DESCRIPTION
**Pull request by ducdetronquito**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -77,7 +77,7 @@ via one of their parent class.
 
         @property
         def roar(self) -> str:
-            return "Grrr!"
+            return "Grrr! 🦁"
```
Comment (pewho) [inline README.rst:80]: !important

File: nurse/api.py [hunk 2]
```diff
@@ -15,29 +53,40 @@ class SSHClient:
...
+    asyncio.run(send())
     """
 
     def decorator(decorated):
         service_catalog = ServiceCatalog.get_instance()
         if isinstance(decorated, type):
             return inject_class(decorated, service_catalog, items_to_inject)
-        elif callable(decorated):
-            return inject_method(decorated, service_catalog, items_to_inject)
+        elif iscoroutinefunction(decorated):
```
Comment (pewho) [inline nurse/api.py:84]: *_* GG

File: nurse/api.py [hunk 3]
```diff
@@ -15,29 +53,40 @@ class SSHClient:
...
+            return inject_async_function(decorated, service_catalog, items_to_inject)
+        elif isfunction(decorated):
+            return inject_function(decorated, service_catalog, items_to_inject)
 
-        raise NotImplementedError('user-defined class or method can be injected.')
+        raise NotImplementedError("user-defined class or function can be injected.")
```
Comment (sGeeK44) [inline nurse/api.py:89]: can't a la place de can non ?

Comment (ducdetronquito) [reply nurse/api.py]: Fixed :)

**Pull request merged**